### PR TITLE
Issue 18724 - Mounting errors out if host does does not support file system 9p. Detect  and display message. fixes <18724>

### PR DESCRIPTION
**Pull request by xcarolan**
before:

📁  Mounting host path /home into VM as /host ...
    ▪ Mount type:   9p
    ▪ User ID:      docker
    ▪ Group ID:     docker
    ▪ Version:      9p2000.L
    ▪ Message Size: 262144
    ▪ Options:      map[]
    ▪ Bind Address: 192.168.49.1:40009
🚀  Userspace file server: ufs starting

❌  Exiting due to GUEST_MOUNT: mount failed: mount with cmd /bin/bash -c "sudo mount -t 9p -o dfltgid=$(grep ^docker: /etc/group | cut -d: -f3),dfltuid=$(id -u docker),msize=262144,port=40009,trans=tcp,version=9p2000.L 192.168.49.1 /host" : /bin/bash -c "sudo mount -t 9p -o dfltgid=$(grep ^docker: /etc/group | cut -d: -f3),dfltuid=$(id -u docker),msize=262144,port=40009,trans=tcp,version=9p2000.L 192.168.49.1 /host": Process exited with status 32
stdout:

stderr:
mount: /host: unknown filesystem type '9p'.

╭───────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                           │
│    😿  If the above advice does not help, please let us know:                             │
│    👉  https://github.com/kubernetes/minikube/issues/new/choose                           │
│                                                                                           │
│    Please run `minikube logs --file=logs.txt` and attach logs.txt to the GitHub issue.    │
│    Please also attach the following file to the GitHub issue:                             │
│    - /tmp/minikube_mount_88aa2fc65fee468afe71466d6581fa82d76bb8be_0.log                   │
│                                                                                           │
╰───────────────────────────────────────────────────────────────────────────────────────────╯

After:
]
❌  Exiting due to HOST_UNSUPPORTED: The host does not support filesystem 9p

**Build and Test Output**

= make lint =============================================================
ok
= go mod ================================================================
ok
= boilerplate ===========================================================
ok
= schema_check ==========================================================
ok
= go test ===============================================================
ok      k8s.io/minikube/cmd/minikube/cmd        0.288s  coverage: 20.0% of statements
ok      k8s.io/minikube/cmd/minikube/cmd/config 0.147s  coverage: 23.7% of statements
ok      k8s.io/minikube/pkg/addons      0.061s  coverage: 26.2% of statements
ok      k8s.io/minikube/pkg/drivers     0.008s  coverage: 42.3% of statements
ok      k8s.io/minikube/pkg/drivers/hyperkit    0.004s  coverage: 77.3% of statements
ok      k8s.io/minikube/pkg/drivers/kic/oci     6.025s  coverage: 19.1% of statements
ok      k8s.io/minikube/pkg/drivers/kvm 0.021s  coverage: 4.1% of statements
ok      k8s.io/minikube/pkg/minikube/assets     0.033s  coverage: 29.0% of statements
ok      k8s.io/minikube/pkg/minikube/audit      0.022s  coverage: 77.0% of statements
ok      k8s.io/minikube/pkg/minikube/bootstrapper       0.811s  coverage: 42.1% of statements
ok      k8s.io/minikube/pkg/minikube/bootstrapper/bsutil        0.068s  coverage: 63.9% of statements
ok      k8s.io/minikube/pkg/minikube/bootstrapper/bsutil/ktmpl  0.002s  coverage: 100.0% of statements
ok      k8s.io/minikube/pkg/minikube/bootstrapper/images        0.017s  coverage: 73.8% of statements
ok      k8s.io/minikube/pkg/minikube/cluster    0.054s  coverage: 12.0% of statements
ok      k8s.io/minikube/pkg/minikube/cni        0.041s  coverage: 8.7% of statements
ok      k8s.io/minikube/pkg/minikube/command    0.029s  coverage: 11.7% of statements
ok      k8s.io/minikube/pkg/minikube/config     0.025s  coverage: 67.0% of statements
ok      k8s.io/minikube/pkg/minikube/cruntime   0.044s  coverage: 29.3% of statements
ok      k8s.io/minikube/pkg/minikube/docker     0.030s  coverage: 20.8% of statements
ok      k8s.io/minikube/pkg/minikube/download   1.032s  coverage: 27.2% of statements
ok      k8s.io/minikube/pkg/minikube/driver     0.017s  coverage: 44.2% of statements
ok      k8s.io/minikube/pkg/minikube/driver/auxdriver   0.024s  coverage: 19.0% of statements
ok      k8s.io/minikube/pkg/minikube/extract    0.004s  coverage: 56.9% of statements
ok      k8s.io/minikube/pkg/minikube/image      0.016s  coverage: 5.4% of statements
ok      k8s.io/minikube/pkg/minikube/kubeconfig 0.041s  coverage: 78.3% of statements
ok      k8s.io/minikube/pkg/minikube/localpath  0.005s  coverage: 47.4% of statements
ok      k8s.io/minikube/pkg/minikube/logs       0.038s  coverage: 5.4% of statements
ok      k8s.io/minikube/pkg/minikube/machine    0.082s  coverage: 20.3% of statements
ok      k8s.io/minikube/pkg/minikube/mustload   0.061s  coverage: 6.4% of statements
ok      k8s.io/minikube/pkg/minikube/notify     0.061s  coverage: 79.8% of statements
ok      k8s.io/minikube/pkg/minikube/out        0.035s  coverage: 68.2% of statements
ok      k8s.io/minikube/pkg/minikube/out/register       0.016s  coverage: 55.4% of statements
ok      k8s.io/minikube/pkg/minikube/perf       4.024s  coverage: 20.7% of statements
ok      k8s.io/minikube/pkg/minikube/proxy      0.023s  coverage: 74.0% of statements
ok      k8s.io/minikube/pkg/minikube/reason     0.017s  coverage: 70.0% of statements
ok      k8s.io/minikube/pkg/minikube/registry   0.028s  coverage: 78.8% of statements
ok      k8s.io/minikube/pkg/minikube/registry/drvs/docker       0.050s  coverage: 49.6% of statements
ok      k8s.io/minikube/pkg/minikube/service    5.124s  coverage: 76.6% of statements
ok      k8s.io/minikube/pkg/minikube/shell      0.006s  coverage: 94.4% of statements
ok      k8s.io/minikube/pkg/minikube/storageclass       0.017s  coverage: 100.0% of statements
ok      k8s.io/minikube/pkg/minikube/style      0.003s  coverage: 100.0% of statements
ok      k8s.io/minikube/pkg/minikube/sysinit    0.020s  coverage: 5.9% of statements
ok      k8s.io/minikube/pkg/minikube/translate  0.055s  coverage: 27.6% of statements
ok      k8s.io/minikube/pkg/minikube/tunnel     1.515s  coverage: 63.8% of statements
ok      k8s.io/minikube/pkg/network     0.004s  coverage: 75.5% of statements
ok      k8s.io/minikube/pkg/util        0.362s  coverage: 79.9% of statements
ok      k8s.io/minikube/pkg/util/lock   0.003s  coverage: 10.0% of statements
ok      k8s.io/minikube/pkg/util/retry  0.003s  coverage: 0.0% of statements
ok

### CODE REVIEW

**Comment by medyagh** [comment]
Thank you @xcarolan for this PR, could plz specify what OS did you try it on for the Before/After ?

**Comment by medyagh** [comment]
thank you @xcarolan for this PR, is the before/after for Linux, mind sharing what Linux you tried it on ?

**Comment by xcarolan** [comment]
Hey Medyagh -yes of course. I tried it on Ubuntu 24.04 and RHEL 9.

File: go.mod [hunk 1]
```diff
@@ -200,6 +203,7 @@ require (
 	github.com/prometheus/procfs v0.10.1 // indirect
 	github.com/prometheus/prometheus v0.35.0 // indirect
 	github.com/rivo/uniseg v0.2.0 // indirect
+	github.com/rogpeppe/go-internal v1.10.0 // indirect
 	github.com/russross/blackfriday/v2 v2.1.0 // indirect
 	github.com/sagikazarmark/locafero v0.4.0 // indirect
 	github.com/sagikazarmark/slog-shim v0.1.0 // indirect
```
Comment (spowelljr) [inline go.mod:209]: No new modules were added so this shouldn't change, could you revert this file

File: cmd/minikube/cmd/mount.go [hunk 2]
```diff
@@ -186,6 +186,12 @@ var mountCmd = &cobra.Command{
 			cfg.Options[parts[0]] = parts[1]
 		}
 
+		// check if linux host supports file system 9p
+		if runtime.GOOS == "linux" {
+			if !detect.IsNinePSupported() {
```
Comment (spowelljr) [inline cmd/minikube/cmd/mount.go:191]: Could we combine these ifs with `&&` since there's no else case for the nested if

**Review by spowelljr — CHANGES_REQUESTED** [review]

**Comment by xcarolan** [comment]
closing

**Pull request closed without merging**